src/hardware/pcspeaker.cpp

```diff
@@ -18,532 +17,70 @@
+		        model_pref.data());
 		return;
-	case 2:
-		while (passed>0) {
-			/* passed the initial low cycle? */
-			if (spkr.pit_index>=spkr.pit_half) {
-				/* Start a new low cycle */
-				if ((spkr.pit_index+passed)>=spkr.pit_max) {
-					const auto delay = spkr.pit_max -
-					                   spkr.pit_index;
-					delay_base+=delay;passed-=delay;
-					spkr.pit_last = AMPLITUDE_NEGATIVE;
-					if (spkr.mode==SPKR_PIT_ON) AddDelayEntry(delay_base,spkr.pit_last);
-					spkr.pit_index=0;
-				} else {
-					spkr.pit_index+=passed;
-					return;
-				}
-			} else {
-				if ((spkr.pit_index+passed)>=spkr.pit_half) {
-					const auto delay = spkr.pit_half -
-					                   spkr.pit_index;
-					delay_base+=delay;passed-=delay;
-					spkr.pit_last = AMPLITUDE_POSITIVE;
-					if (spkr.mode==SPKR_PIT_ON) AddDelayEntry(delay_base,spkr.pit_last);
-					spkr.pit_index=spkr.pit_half;
-				} else {
-					spkr.pit_index+=passed;
-					return;
-				}
-			}
-		}
-		break;
-		//END CASE 2
-	case 3:
-		while (passed>0) {
-			/* Determine where in the wave we're located */
-			if (spkr.pit_index>=spkr.pit_half) {
-				if ((spkr.pit_index+passed)>=spkr.pit_max) {
-					const auto delay = spkr.pit_max -
-					                   spkr.pit_index;
-					delay_base+=delay;passed-=delay;
-					spkr.pit_last = AMPLITUDE_POSITIVE;
-					if (spkr.mode==SPKR_PIT_ON) AddDelayEntry(delay_base,spkr.pit_last);
-					spkr.pit_index=0;
-					/* Load the new count */
-					spkr.pit_half=spkr.pit_new_half;
-					spkr.pit_max=spkr.pit_new_max;
-				} else {
-					spkr.pit_index+=passed;
-					return;
-				}
-			} else {
-				if ((spkr.pit_index+passed)>=spkr.pit_half) {
-					const auto delay = spkr.pit_half -
-					                   spkr.pit_index;
-					delay_base+=delay;passed-=delay;
-					spkr.pit_last = AMPLITUDE_NEGATIVE;
-					if (spkr.mode==SPKR_PIT_ON) AddDelayEntry(delay_base,spkr.pit_last);
-					spkr.pit_index=spkr.pit_half;
-					/* Load the new count */
-					spkr.pit_half=spkr.pit_new_half;
-					spkr.pit_max=spkr.pit_new_max;
-				} else {
-					spkr.pit_index+=passed;
-					return;
-				}
-			}
-		}
-		break;
-		//END CASE 3
-	case 4:
-		if (spkr.pit_index<spkr.pit_max) {
-			/* Check if we're gonna pass the end this block */
-			if (spkr.pit_index+passed>=spkr.pit_max) {
-				const auto delay = spkr.pit_max - spkr.pit_index;
-				delay_base+=delay;passed-=delay;
-				spkr.pit_last = AMPLITUDE_NEGATIVE;
-				if (spkr.mode==SPKR_PIT_ON) AddDelayEntry(delay_base,spkr.pit_last);				//No new events unless reprogrammed
-				spkr.pit_index=spkr.pit_max;
-			} else spkr.pit_index+=passed;
-		}
-		break;
-		//END CASE 4
 	}
-}
 
-// PIT-mode activation
-void PCSPEAKER_SetCounter(int cntr, int mode)
-{
-	if (!SpeakerExists())
-		return;
+	// Get the user's filering preference
+	const auto filter_pref  = std::string_view(prop->Get_string("pcspeaker_filter"));
+	const auto filter_state = filter_pref == "on" ? FilterState::On : FilterState::Off;
```

Comment:
Maybe copy my way of parsing the filter value and emit a warning for invalid strings? (e.g. check out in `gameblaster.cpp`)

Reply:
worked like a charm, thanks!